### PR TITLE
chore: refactor performance test to generate queries and data together

### PR DESCRIPTION
**Pull request by lesam**
Initial refactoring in preparation for https://github.com/influxdata/influxdb/issues/22872

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

**Comment by lesam** [comment]
See test job here: https://app.circleci.com/pipelines/github/influxdata/influxdb/25759/workflows/3ff3240a-1e66-4c85-aa9d-73b3284b5583/jobs/229339

File: .circleci/config.yml
Comment (lesam) [inline .circleci/config.yml:33]: We're keeping the circleci perf job running while the AWS instance is up, so we ditch the teardown job and the complicated configuration around that.

File: .circleci/config.yml
Comment (lesam) [inline .circleci/config.yml:101]: - [ ] Fix the fixme before merging - this is to test out the job before merge.

File: .circleci/config.yml
Comment (lesam) [inline .circleci/config.yml:131]: flux and influxql are now tested during the same script run.

File: .circleci/config.yml
Comment (lesam) [inline .circleci/config.yml:725]: Replacement for the teardown job.

File: scripts/ci/perf_test.sh
Comment (lesam) [inline scripts/ci/perf_test.sh:30]: Instead of pulling influx_tools to the circleci machine and then SCP'ing to the performance test machine, just pull it directly to the test machine.

File: scripts/ci/perf_test.sh
Comment (lesam) [inline scripts/ci/perf_test.sh:41]: The old circleci machine executor appears to have had StrictHostKeyChecking=no in the machine definition. The new docker instance (correctly) does not, so we prime the pump here in order to save the host key.

File: scripts/ci/perf_test.sh
Comment (lesam) [inline scripts/ci/perf_test.sh:64]: Move this into the run_perftest script

File: scripts/ci/perf_test.sh
Comment (lesam) [inline scripts/ci/perf_test.sh:76]: moved this into the run_perftest script

File: scripts/ci/perf_test.sh
Comment (lesam) [inline scripts/ci/perf_test.sh:6]: Bump the version here instead of trying to slam in `libc6` during `run_perftest.sh`.

File: scripts/ci/run_perftest.sh
Comment (lesam) [inline scripts/ci/run_perftest.sh:12]: We bumped the ubuntu version of the AMI, so we're good on this now.

File: scripts/ci/run_perftest.sh
Comment (lesam) [inline scripts/ci/run_perftest.sh:21]: No longer necessary - the log is shown in circleci instead of being copied to S3

File: scripts/ci/run_perftest.sh
Comment (lesam) [inline scripts/ci/run_perftest.sh:22]: No longer necessary - circleci tears down the instance after `run_perftest.sh` exits.

File: scripts/ci/run_perftest.sh
Comment (lesam) [inline scripts/ci/run_perftest.sh:119]: I think this was actually expected behaviour - anyway, it works now.

File: scripts/ci/run_perftest.sh
Comment (lesam) [inline scripts/ci/run_perftest.sh:227]: Note `group-window-transpose`, `cardinality` usecases that didn't actually ever get passed in, since those aren't real query usecases.

File: scripts/ci/run_perftest.sh
Comment (lesam) [inline scripts/ci/run_perftest.sh:300]: test both http and flux-http

File: scripts/ci/run_perftest.sh
Comment (lesam) [inline scripts/ci/run_perftest.sh:271]: Should be relatively straightforward to factor out a configuration driven approach, where each test has a data generation strategy, and a list of query strategies.

File: scripts/ci/perf_test.sh
Comment (lesam) [inline scripts/ci/perf_test.sh:31]: We don't look for the latest anymore, I just hard-code a particular version to pull from S3 into this script.

File: .circleci/config.yml
Comment (wbaker85) [reply .circleci/config.yml:33]: This seems like a nice simplification. It'll take a much longer time for the job to finish now obviously but since they only run on master it's not like it is going to be blocking anything.

File: .circleci/config.yml
Comment (wbaker85) [reply .circleci/config.yml:131]: We had split these up in an effort to parallelize and run things faster...but like my comment above alludes to; it shouldn't really matter since they only run on master.

If we ever want to run these on PRs and possibly gate PRs based on the results, we'll need to figure out a way to make them go faster, but that's a totally different discussion.

File: .circleci/config.yml
Comment (wbaker85) [inline .circleci/config.yml:695]: Is `record_results` used for anything other than CI update dry runs?

File: scripts/ci/run_perftest.sh
Comment (wbaker85) [inline scripts/ci/run_perftest.sh:20]: I guess this must work based on the dry run, but I know that sometimes there's a period of time between when the `influxd` process is started successfully and when it can actually respond to http requests. Does `systemctl start influxdb` not return until `influxd` is actually able to respond to http requests?

File: scripts/ci/run_perftest.sh
Comment (wbaker85) [inline scripts/ci/run_perftest.sh:120]: On lines 118/119, do we even need to clone the comparisons repo?

**Review by DStrand1 — COMMENTED** [review]
LGTM from what I can tell, though @wbaker85 should review as he would know better than I if there were issues

File: .circleci/config.yml
Comment (lesam) [reply .circleci/config.yml:695]: nope

File: scripts/ci/run_perftest.sh
Comment (lesam) [reply scripts/ci/run_perftest.sh:20]: I believe the start script tries to wait for influxdb to actually start, yes.

File: scripts/ci/run_perftest.sh
Comment (lesam) [reply scripts/ci/run_perftest.sh:120]: I don't think so - nice!

- [ ] Try removing the 'git clone' above for influxdb-comparisons

File: scripts/ci/run_perftest.sh
Comment (wbaker85) [reply scripts/ci/run_perftest.sh:20]: Oh, yep, it definitely does: https://github.com/influxdata/influxdb/blob/fa9ba8e86f4f21344cd1edd66fc8bed07331eb10/scripts/influxd-systemd-start.sh#L23-L28

File: .circleci/config.yml
Comment (lesam) [reply .circleci/config.yml:131]: I think ideally we would be able to spin up separate AWS instances for each category of tests or something similar - ideally the split would be based on dataset rather than by [flux vs influxql]. Before I merged these back together we had to pay the cost of installing each dataset twice.

For now I value simplicity, but in the future yeah it would be great to split up and run multiple tests in parallel. I think having this configuration driven will make that easier.

File: scripts/ci/run_perftest.sh
Comment (wbaker85) [inline scripts/ci/run_perftest.sh:344]: Another thought: If the results aren't being recorded, instead of totally omitting this step it would be nice to instead run it with the `--test` flag: https://github.com/influxdata/telegraf#run-a-single-telegraf-collection-outputting-metrics-to-stdout so that we can see what would be sent to the cloud dashboard in the logs.

I usually just let it report the results anyway on testing branches, since the data on the dashboards is filtered by branch. Having the test output in the logs might make it even easier though, that way you wouldn't have to go back into the cloud dashboards and clean up the extra data.

**Comment by lesam** [comment]
Ready for review and merge.

**Review by wbaker85 — APPROVED** [review]

**Pull request merged**
